internal/pkg/cli/pipeline_init.go

```diff
@@ -283,12 +285,24 @@ func (o *initPipelineOpts) askGitHubRepoDetails() error {
 	o.repoName = repoDetails.name
 	o.repoOwner = repoDetails.owner
 
-	if o.repoBranch == "" {
-		o.repoBranch = defaultGHBranch
-	}
 	return nil
 }
 
+// getBranch fetches the user's current branch as a best-guess of which branch they want their pipeline to follow. If err, insert default branch name.
+func (o *initPipelineOpts) getBranch() {
+	// Fetches local git branch.
+	err := o.runner.Run("git", []string{"rev-parse", "--abbrev-ref", "HEAD"}, exec.Stdout(&o.buffer))
+	o.repoBranch = strings.TrimSpace(o.buffer.String())
+	if err != nil {
+		o.repoBranch = defaultBranch
+	}
+	if strings.TrimSpace(o.buffer.String()) == "" {
+		o.repoBranch = defaultBranch
+	}
+	o.buffer.Reset()
+	log.Infof("Your pipeline will follow branch '%s'. You may make changes in the pipeline manifest before deployment.\n", color.HighlightUserInput(o.repoBranch))
```

Comment:
nit: (just to be within 80 chars for the default terminal screen size)
```suggestion
	log.Infof(`Your pipeline will follow branch '%s'. 
You may make changes in the pipeline manifest before deployment.
`, color.HighlightUserInput(o.repoBranch))
```